frontend/public/components/dashboards-page/overview-dashboard/top-consumers-card.tsx

```diff
@@ -0,0 +1,137 @@
+    };
+  });
+  return topConsumers;
+};
+
+const TopConsumersCard_: React.FC<DashboardItemProps> = ({
+  watchPrometheus,
+  stopWatchPrometheusQuery,
+  prometheusResults,
+}) => {
+  const [type, setType] = React.useState(PODS);
+  const [sortOption, setSortOption] = React.useState(MetricType.CPU);
+
+  React.useEffect(() => {
+    const topConsumersMap = getTopConsumersQueries();
+    const currentQuery = topConsumersMap[type].queries[sortOption];
+    watchPrometheus(currentQuery);
+    return () => stopWatchPrometheusQuery(currentQuery);
+  }, [watchPrometheus, stopWatchPrometheusQuery, type, sortOption]);
+
+  const topConsumersMap = getTopConsumersQueries();
+  const currentQuery = topConsumersMap[type].queries[sortOption];
+  const topConsumersResult = prometheusResults.getIn([currentQuery, 'result']);
+
+  const stats = getInstantVectorStats(topConsumersResult, topConsumersMap[type].metric, metricTypeMap[sortOption].humanize);
+
+  const data = topConsumersMap[type].mutator ? topConsumersMap[type].mutator(stats) : stats;
+
+  return (
+    <DashboardCard>
+      <DashboardCardHeader>
+        <DashboardCardTitle>Top Consumers</DashboardCardTitle>
+      </DashboardCardHeader>
+      <DashboardCardBody>
+        <ConsumersFilter>
+          <Dropdown
+            buttonClassName="co-overview-consumers__dropdown"
+            dropDownClassName="co-overview-consumers__dropdown"
+            className="btn-group co-overview-consumers__dropdown co-overview-consumers__dropdown--right"
+            items={_.mapValues(topConsumersMap, (v, key) => key)}
+            onChange={(v: string) => {
+              setType(v);
+              if (!topConsumersMap[v].queries[sortOption]) {
+                setSortOption(Object.keys(topConsumersMap[v].queries)[0] as MetricType);
+              }
+            }}
+            selectedKey={type}
+            title={type}
+          />
+          <Dropdown
+            className="btn-group co-overview-consumers__dropdown co-overview-consumers__dropdown--left"
+            buttonClassName="co-overview-consumers__dropdown"
+            dropDownClassName="co-overview-consumers__dropdown"
+            items={_.mapValues(topConsumersMap[type].queries, (v, key) => key)}
+            onChange={(v: MetricType) => setSortOption(v)}
+            selectedKey={sortOption}
+            title={sortOption}
+          />
+        </ConsumersFilter>
+        <ConsumersBody isLoading={!topConsumersResult}>
+          <BarChart
+            data={data}
+            query={currentQuery}
+            title={`${type} by ${metricTypeMap[sortOption].description}`}
+          />
+        </ConsumersBody>
+      </DashboardCardBody>
+    </DashboardCard>
+  );
+};
+
+export const TopConsumersCard = withDashboardResources(TopConsumersCard_);
+
+export type ConsumerMutator = (data: DataPoint[]) => DataPoint[];
+
+type TopConsumersMap = {
+  [key: string]: {
+    metric: string,
+    queries: { [key in MetricType]?: string },
```

Comment:
Why `[key in MetricType]?` with the trailing `?`

Reply:
If I dont add `?` then TS is expecting all values of `MetricType` in `queries` prop and thats not what I always want ie here https://github.com/openshift/console/pull/1722/files#diff-163991d4c119db922a729100ce2bea7fR27 i dont have `MetricType.NETWORK`

Reply:
OK, makes sense.

Reply:
Actually, we could re-use `DashboardsOverviewTopConsumerItem.queries` type here, but we can improve it later.